### ISSUE TITLE
[NavList.Item] - Redundant use of aria-labelledby for accessible name

### ISSUE THREAD

**Issue by garyb1** [primer/react#4029]
Hi, 
I have a couple of questions/ideas about the [NavList.Item component](https://primer.style/react/NavList). 

1. I think there might be redundant use of aria-labelledby in this component. This is a minor thing, but kind of goes against the [first rule of ARIA](https://www.w3.org/TR/using-aria/#rule1).
2. You could [enforce the accessibility](https://adrianroselli.com/2021/06/using-css-to-enforce-accessibility.html) of the NavList.Item component through CSS. I really like this pattern.
3. Is there a reason why the tabindex is explicitly set on the anchor element?

Curious to get your thoughts on the above. I would love to contribute if at all possible.
Thanks.

**Comment by garyb1** [comment]
I just noticed the contributing document says
>  Please note that at this time, we are not looking for external contributions from non-GitHub staff. Some of the links included in these documents are only available to GitHub staff.

**Comment by siddharthkp** [comment]
Hi!

Thanks for your feedback! Addressing some of it here:

> 1. I think there might be redundant use of aria-labelledby in this component. This is a minor thing, but kind of goes against the [first rule of ARIA](https://www.w3.org/TR/using-aria/#rule1). The text content of the anchor element should be fine for the accessible name.

That's a very fair critique, it's wired this way to take care of just one use case: trailing supplementary text in the nav item that is not part of the label. Example: https://primer.style/react/NavList#with-trailing-visuals

![NavList with trailing text](https://github.com/primer/react/assets/1863771/9961d9fe-59d5-4fb5-8ebb-dd0ab7116f46)

> 3. Is there a reason why the tabindex is explicitly set on the anchor element?

No, we don't need it at all. It's coming from silly implementation detail, that's deeply nested (NavList.Item → ActionList.LinkItem → ActionList.Item), I can look into removing it.

**Comment by garyb1** [comment]
Hey @siddharthkp, thanks for the reply.

Sorry I didn't see the example with trailing visuals. Thanks for sharing it. I think it would be nice to reference that trailing supplementary text with `aria-describedby` perhaps? And maybe some visually hidden text like the word "items" so it would read "1,234 items"? Although I do see in the code `aria-describedby` is used for block descriptions, so I'm not so sure. 

If I can help with anything do let me know.

**Comment by siddharthkp** [comment]
> I think it would be nice to reference that trailing supplementary text with aria-describedby perhaps? 
> Although I do see in the code aria-describedby is used for block descriptions, so I'm not so sure.

Yeah, I agree with you. I think NavList doesn't have description yet, so it might be okay.

Let me look into the different ways its used and if it's safe to use it as description

**Comment by siddharthkp** [comment]
I'm going to close this issue for now, if that's okay with you?

**Comment by garyb1** [comment]
Cool, no problem @siddharthkp 👍
